credentials/insecure/insecure.go

```diff
@@ -74,25 +74,29 @@ func (info) AuthType() string {
 // insecureBundle implements an insecure bundle.
 // An insecure bundle provides a thin wrapper around insecureTC to support
 // the credentials.Bundle interface.
-type insecureBundle struct{}
+type insecureBundle struct {
+	tc credentials.TransportCredentials
+}
```

Comment:
I'm sorry, but this seems worse.  It adds state for zero benefit.  If Go has (or might one day have) optimizations for empty structs, this defeats those optimizations.

Reply:
Changed back to the previous version.